### PR TITLE
Move range logic into range utils and remove range-specific statuses from CacheEntryStatus

### PR DESCRIPTION
**Pull request by capoferro**
Commit Message: Move range logic into range utils and remove range-specific statuses from CacheEntryStatus
Risk Level: Low (refactor)
Testing: Existing

### CODE REVIEW

File: source/extensions/filters/http/cache/cache_filter.cc [hunk 1]
```diff
@@ -234,64 +235,39 @@ void CacheFilter::onHeaders(LookupResult&& result, Http::RequestHeaderMap& reque
...
     return;
   }
+
   // TODO(yosrym93): Handle request only-if-cached directive
-  switch (result.cache_entry_status_) {
+  lookup_result_ = std::make_unique<LookupResult>(std::move(result));
+  switch (lookup_result_->cache_entry_status_) {
   case CacheEntryStatus::FoundNotModified:
-    NOT_IMPLEMENTED_GCOVR_EXCL_LINE; // We don't yet return or support these codes.
+    // TODO(b/194204291): Implement proper handling of FoundNotModified. Until
```
[the remark this answers is not in the document]
Comment (capoferro) [reply source/extensions/filters/http/cache/cache_filter.cc:243]: Yep, also can be moved to a new PR.

File: source/extensions/filters/http/cache/cache_headers_utils.h [hunk 2]
```diff
@@ -54,12 +55,17 @@ struct RequestCacheControl {
   //   now - expiration_time < max-stale
   // If max-stale has no value then the client is willing to receive any stale response
   OptionalDuration max_stale_;
+
+private:
+  void applyCacheControlHeader(absl::string_view cache_control_header);
+  void applyPragmaHeader(absl::string_view pragma_header);
```
Comment (RyanTheOptimist) [inline source/extensions/filters/http/cache/cache_headers_utils.h:61]: Whoops, I think the changes to the .cc file were moved of to a different PR, so the .h changes probably should be too.

**Review by RyanTheOptimist — APPROVED** [review]
Thanks for the cleanups! This looks great, modulo moving the changes in cache_headers_utils.h to the same PR that the changes in cache_headers_utils.h moved to. I'll approve and you can do that without waiting for another round of review.

(Feel free to send that PR my way if you're looking for a reviewer)

File: source/extensions/filters/http/cache/cache_headers_utils.h (hunk 2 above)
Comment (capoferro) [reply source/extensions/filters/http/cache/cache_headers_utils.h]: Oops....

**Comment by jmarantz** [comment]
Can you re-add me to the PR once Ryan's OK with hit and it's passing CI? Thanks!

**Review by RyanTheOptimist — APPROVED** [review]
Argh! I meant to approve yesterday, but missed hitting the button. That being said, it looks like format is still red:

Format check failed, try apply following patch to fix:
diff --git a/test/extensions/filters/http/cache/cache_filter_test.cc b/test/extensions/filters/http/cache/cache_filter_test.cc
index 9425d83ba..153a20f03 100644
--- a/test/extensions/filters/http/cache/cache_filter_test.cc
+++ b/test/extensions/filters/http/cache/cache_filter_test.cc
@@ -137,7 +137,8 @@ protected:
 TEST_F(CacheFilterTest, FilterIsBeingDestroyed) {
   CacheFilterSharedPtr filter = makeFilter(simple_cache_);
   filter->onDestroy();
-  filter->decodeHeaders(request_headers_, true); // Should do nothing... at least make sure it doesn't crash.
+  filter->decodeHeaders(request_headers_,
+                        true); // Should do nothing... at least make sure it doesn't crash.
 }

**Comment by capoferro** [comment]
> Argh! I meant to approve yesterday, but missed hitting the button. That being said, it looks like format is still red:

Maybe it discards your approval when I push changes? I thought you approved yesterday, as well.

**Comment by RyanTheOptimist** [comment]
> > Argh! I meant to approve yesterday, but missed hitting the button. That being said, it looks like format is still red:
> 
> Maybe it discards your approval when I push changes? I thought you approved yesterday, as well.

Oh, I think you're right. My approval seems to be gone again. But looking back at email, I do see
```
@RyanTheOptimist approved this pull request
```
two different times. Good to know! LOL

**Comment by zuercher** [comment]
This is failing tests because coverage dropped below the threshold:

```
Code coverage for source/extensions/filters/http/cache is lower than limit of 92.6 (92.0)
```
